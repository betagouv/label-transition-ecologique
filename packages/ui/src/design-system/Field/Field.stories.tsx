import {Meta, StoryObj} from '@storybook/react';

import {Field} from '.';

const meta: Meta<typeof Field> = {
  component: Field,
  decorators: [story => <div className="flex">{story()}</div>],
};

export default meta;

type Story = StoryObj<typeof Field>;

const FakeInput = () => (
  <input
    type="text"
    className="border border-solid border-grey-4 rounded-lg min-h-[3rem] px-4 placeholder:text-sm"
    placeholder="Placeholder"
  />
);

export const Default: Story = {
  args: {title: "Description de l'action"},
  render: args => (
    <div className="flex flex-col gap-6 w-full max-w-sm">
      <Field {...args}>
        <FakeInput />
      </Field>
      <Field {...args} title="Small" small>
        <FakeInput />
      </Field>
    </div>
  ),
};

export const DisabledWithHint: Story = {
  args: {
    title: "Description de l'action",
    state: 'disabled',
    hint: 'Texte description additionnel',
    message: 'Message d’information',
  },
  render: args => (
    <Field {...args}>
      <FakeInput />
    </Field>
  ),
};

export const WithHint: Story = {
  args: {
    title: "Description de l'action",
    hint: 'Texte description additionnel',
    message: 'Message d’information',
  },
  render: args => (
    <Field {...args}>
      <FakeInput />
    </Field>
  ),
};

export const WithInfo: Story = {
  args: {
    title: "Description de l'action",
    state: 'info',
    message: 'Message d’information',
  },
  render: args => (
    <Field {...args}>
      <FakeInput />
    </Field>
  ),
};

export const WithError: Story = {
  args: {
    title: "Description de l'action",
    state: 'error',
    message: "Message d'erreur",
  },
  render: args => (
    <Field {...args}>
      <FakeInput />
    </Field>
  ),
};

export const WithSuccess: Story = {
  args: {
    title: "Description de l'action",
    state: 'success',
    message: 'Message de succès',
  },
  render: args => (
    <Field {...args}>
      <FakeInput />
    </Field>
  ),
};

export const WithWarning: Story = {
  args: {
    title: "Description de l'action",
    state: 'warning',
    message: "Message d'avertissement",
  },
  render: args => (
    <Field {...args}>
      <FakeInput />
    </Field>
  ),
};
