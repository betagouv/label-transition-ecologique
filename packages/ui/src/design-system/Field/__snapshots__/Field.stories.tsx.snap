// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`design-system/Field Default smoke-test 1`] = `
<div class="flex">
  <div class="flex-grow flex flex-col gap-3">
    <label for="un_nom"
           class="ml-0 mb-0"
    >
      <div class="font-medium text-grey-8">
        Description de l'action
      </div>
    </label>
    <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
      <input type="text"
             class="grow text-grey-8 px-4 outline-none text-md py-3"
             id="un_nom"
             placeholder="Placeholder"
      >
    </div>
  </div>
</div>
`;

exports[`design-system/Field DisabledWithHint smoke-test 1`] = `
<div class="flex">
  <div class="flex-grow flex flex-col gap-3">
    <label for="un_nom"
           class="ml-0 mb-0"
    >
      <div class="font-medium text-grey-8 text-grey-5">
        Description de l'action
      </div>
      <span class="block mt-2 text-xs text-grey-7 text-grey-5">
        Texte description additionnel
      </span>
    </label>
    <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
      <input type="text"
             class="grow text-grey-8 px-4 outline-none text-md py-3"
             id="un_nom"
             placeholder="Placeholder"
             disabled
      >
    </div>
    <div class="flex items-center text-grey-6">
      <span class="text-xs">
        Message d’information
      </span>
    </div>
  </div>
</div>
`;

exports[`design-system/Field WithError smoke-test 1`] = `
<div class="flex">
  <div class="flex-grow flex flex-col gap-3">
    <label for="un_nom"
           class="ml-0 mb-0"
    >
      <div class="font-medium text-grey-8">
        Description de l'action
      </div>
    </label>
    <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-error-1">
      <input type="text"
             class="grow text-grey-8 px-4 outline-none text-md py-3"
             id="un_nom"
             placeholder="Placeholder"
      >
    </div>
    <div class="flex items-center text-error-1">
      <span class="ri-spam-fill font-normal text-center h-4 w-4 text-[1rem] leading-[1rem] mr-1">
      </span>
      <span class="text-xs">
        Message d'erreur
      </span>
    </div>
  </div>
</div>
`;

exports[`design-system/Field WithHint smoke-test 1`] = `
<div class="flex">
  <div class="flex-grow flex flex-col gap-3">
    <label for="un_nom"
           class="ml-0 mb-0"
    >
      <div class="font-medium text-grey-8">
        Description de l'action
      </div>
      <span class="block mt-2 text-xs text-grey-7">
        Texte description additionnel
      </span>
    </label>
    <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
      <input type="text"
             class="grow text-grey-8 px-4 outline-none text-md py-3"
             id="un_nom"
             placeholder="Placeholder"
      >
    </div>
    <div class="flex items-center text-grey-7">
      <span class="text-xs">
        Message d’information
      </span>
    </div>
  </div>
</div>
`;

exports[`design-system/Field WithInfo smoke-test 1`] = `
<div class="flex">
  <div class="flex-grow flex flex-col gap-3">
    <label for="un_nom"
           class="ml-0 mb-0"
    >
      <div class="font-medium text-grey-8">
        Description de l'action
      </div>
    </label>
    <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-info-1">
      <input type="text"
             class="grow text-grey-8 px-4 outline-none text-md py-3"
             id="un_nom"
             placeholder="Placeholder"
      >
    </div>
    <div class="flex items-center text-info-1">
      <span class="ri-information-fill font-normal text-center h-4 w-4 text-[1rem] leading-[1rem] mr-1">
      </span>
      <span class="text-xs">
        Message d’information
      </span>
    </div>
  </div>
</div>
`;

exports[`design-system/Field WithSuccess smoke-test 1`] = `
<div class="flex">
  <div class="flex-grow flex flex-col gap-3">
    <label for="un_nom"
           class="ml-0 mb-0"
    >
      <div class="font-medium text-grey-8">
        Description de l'action
      </div>
    </label>
    <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-success-1">
      <input type="text"
             class="grow text-grey-8 px-4 outline-none text-md py-3"
             id="un_nom"
             placeholder="Placeholder"
      >
    </div>
    <div class="flex items-center text-success-1">
      <span class="ri-checkbox-circle-fill font-normal text-center h-4 w-4 text-[1rem] leading-[1rem] mr-1">
      </span>
      <span class="text-xs">
        Message de succès
      </span>
    </div>
  </div>
</div>
`;

exports[`design-system/Field WithWarning smoke-test 1`] = `
<div class="flex">
  <div class="flex-grow flex flex-col gap-3">
    <label for="un_nom"
           class="ml-0 mb-0"
    >
      <div class="font-medium text-grey-8">
        Description de l'action
      </div>
    </label>
    <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-warning-1">
      <input type="text"
             class="grow text-grey-8 px-4 outline-none text-md py-3"
             id="un_nom"
             placeholder="Placeholder"
      >
    </div>
    <div class="flex items-center text-warning-1">
      <span class="ri-alert-fill font-normal text-center h-4 w-4 text-[1rem] leading-[1rem] mr-1">
      </span>
      <span class="text-xs">
        Message d'avertissement
      </span>
    </div>
  </div>
</div>
`;
